### ISSUE TITLE
Repo: Add an end-to-end (Playwright) test for the playground's AST viewer

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#6086]
### Suggestion

Following #6080, I realized there was one more test for the https://typescript-eslint.io/play playground that I wanted to add!

1. Type some code into the editor
2. Enable the AST viewer
3. Check that the AST shows nodes corresponding to the code

Leaving this open as a `good first issue` so anybody who wants to contribute can. 

<img width="1302" alt="Screenshot of typescript-eslint.io showing the ESTree AST for console.log('hello, world!')" src="https://user-images.githubusercontent.com/3335181/203917613-1764e0a3-5588-4c83-96aa-32959258e866.png">

**Comment by armano2** [comment]
some tests that we should add:


### Type Checking: ☑️

/play/#ts=4.7.1-rc&sourceType=module&code=PTAEFpK6ZhYAUCUBnALgJwJYGM3gCMB7IgGwFMBDAO3HIA8AHDclFLI6lRZWPvxIiwAzUAApMAV3KgAZLNABGUAGolASlABvUAF9EAdwAWWCuOGVSKGQB8boAESUHqxwQeat+hGPoAuUCICACtyPE0AXgA+UClbewsreNB6UAB+WIxpUADE6wBuQSQwflKoHjBqIkJKa3A0avRsagBzCogy0qKHFzUvfNBQZEAZclAqmrqG8CasVqLeTv528eFSIko0WZbwZiIAWyxrbmKOxdgi2oBPahxQYUkbzc47kjFPUBY0SQxqJQHvBiMIgYNB3B54Di-AiUDBvbSIQbCV7qAYjMbVVbrTatHYYfaHVigRDeRAANyIWAAJmNyAZQAAFPEHaxiOHRbR6FFDMCjFZrDZbXH4o5EhCIQHA0H3R6Q0A4FgbciM4XkOFaBEfchfH40unK5mqtkxHS6LnIQCg5JrtdRyNTiUV5VQ0EqmQS3qieejwJiBTjdgbjuKmJKwTLnuhJMJhIo1RqcJx0KBGKAIrqGa6WUaOaaPaBedUHpJrJTwKSYccSQgJSDQFcbqGIeGvlGAEyxhCDSgGShYUE2vUZw2RY2c3OWrs952U+1iqvBmvSxu-IsulViRgBfUEgA8DwA1lUDNQop5dNy817C8XS+Wiiutyz++m11mTeozWBLYxasWikGgTWda3IuTy-JYpDtoMgzxlwoIACSUCmcoKs6D6qh+oDjmwWAtDa04dlBMGJnBBBIY6ipoe655YewuG2kUUFygmoIANqUAANKABAALpIROvbPgaAB04FiCxCGcSR3EYV+P62rERCCQSImkBB6gaoMaLjFeto3hgFZFCI4jkahg5vKe575uAzIriW-oEscximDIYgmauBrmdoZ5adUNnXvZRyIK5KHuW6mgZHEOR3JY1gYVZfm6QFrBBW5lGaHYmTSHFXoJXZg7HGIACEqVme+ubxYctlCgGRSUmEpAwjIIGypQOA4OQjBoCgABqFKUgAwgAQq5BABFm5JUiiiCte1nU9X1Q3BU6oWxeVOWVf5+WIEAA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6ZfKsugIwHs+SAIZN6AD2LQUySnyaoMAbXDYciaND7RIAGhWqsq7JCHx4fAO4BlTswDm6MADNTyRHqPHT5iwDlYALY86o4u8G4enlDelv5mQjxIAPI8AFaI3KGuiPrYAL65ALqROAQk5FQ09MiMLKxMfLRO5kL49rSSfAGUbgpQ6praJVBlpMgU1HQMzGwNtN3IsG4AJh2aCyiOkANaurl4RGMTVdN1c7BMS4irAG5C0H3bGrvDB+XjlVM1M-WNPEJuWj4RocKhMBwYJ6DSAqAp5IA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Parsing comments: ☑️

/play/#ts=5.0.2&sourceType=module&code=PQKgBApgzgNglgOwC5gEQAEkE8AO0DGATnDkgLTTzLAID2ZArgg1BACZkBuAhoVKgC40EQoVqFUYEMACwAKHn5aCKCgAeYALxgATEA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### User defined compiler options ☑️

/play/#ts=4.7.1-rc&jsx=true&sourceType=module&showAST=es&code=GYVwdgxgLglg9mABAMTnRAKAlALkQKQGUANAOgFEAbAUwFtqwpEBvAWAChEvEAnaqEDyQAeACYwAbgD5hAenHSOAXw5A&rules=N4XyA&tsConfig=N4XyA

### Should show error on missing libs ☑️

/play/#ts=4.6.2&sourceType=module&code=CYUwxgNghgTiAEAzArgOzAFwJYHtXwDcctgB5GABRhwFssBnEAClgHMAueJpgSngF4AfIWLA+AHy68BwqrQYgAPERKCePTiuABuIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkqyARhWANoC64YAviF0A

### Should deduce libs base on target ☑️

/play/#ts=5.0.2&jsx=true&sourceType=module&code=MYewdgzgLgBAHjAvDMBTA7jACgJxAWwEsJUAKASgG4BYAKCA&eslintrc=N4KABGBEAOCGBOBnApvSAuKABALgT2mUQGN4BLaHAWiIBsyA7HAejiVUgBpwo2V4A8pTIB7BogxhQECDHgiAVsmI5JkAHTMcJMQDMyAc3ULEYyDwC+3WdFoBXA4wmYA2j1m4CRUhWp1GqjwAutZQ8Ha0RJLSMpCehCTklDSI9EzMDCJUmQzZEbRUsIj8OMgAJlQiwmKwBcQAFrCMaqjyaJYgFkA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkeAhpgObp4VTokDMk4YAviH5A

### Should parse simple eslint config when pasted ☑️

```js
module.exports = {
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: './tsconfig.json'
  },
  plugins: ['@typescript-eslint'],
  rules: {
    "@typescript-eslint/no-non-null-asserted-optional-chain": "error"
  }
}
```

### Additional checks

- verify if ast is updating ☑️
- verify if page is not crashign when goign from - to playground with navigation ☑️
- verify if linting is beeing triggered ☑️
- verify if resizing works ☑️
- verify if changing theme is applied to editor ☑️

**Comment by JoshuaKGoldberg** [comment]
@armano2 I'm filing a new issue 😄 just taking a while to type out... wanted to give #6275 & @tapanprakasht the glory of resolving one!
